apps/frontend/src/libs/components/app/app.tsx

```diff
@@ -13,13 +14,31 @@ import {
 import { AppRoute } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
+	useAppForm,
 	useAppSelector,
 	useEffect,
 	useLocation,
 } from "~/libs/hooks/hooks.js";
 import { actions as userActions } from "~/modules/users/users.js";
 
+import { type Option } from "../select/select.js";
+
+type SelectData = {
+	options: Option<number>[];
+};
```

Comment:
I don't think that creating such a type everywhere we use `Select` is good idea. What is a purpose? Can't it be inferred inside component itself?

Reply:
Well, unfortunately it might be the best I can do. Controlled components are a bit confusing to me, and all of the typescript examples are made like this. also, the name that we pass to Select itself, should be a name of a field in the type that we pass as first generic (SelectData here). Currently I'm trying to do something about it. If you have more experience with controlled components, I'd be grateful to hear how this can be improved.

Reply:
I have no experience with controlled components to be honest, but I'd like to help. So I will clone your branch later and take some time trying to figure it out

Reply:
Not sure why, but I can't pull your last changes. Only til this commit:
![image](https://github.com/user-attachments/assets/0de4f7e8-8bb9-4e02-9886-625a34899dc2)
Maybe the reason is that your branch after that commit have conflicts ? Can you merge main branch into yours and resolve the conflicts, and then I'll try again